test/cmd/diagnostics.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${OS_ROOT}/hack/util.sh"
+source "${OS_ROOT}/hack/cmd_util.sh"
+os::log::install_errexit
+
+# This test validates the diagnostics command
+
+# available diagnostics (2015-12-21):
+# AnalyzeLogs ClusterRegistry ClusterRoleBindings ClusterRoles ClusterRouter ConfigContexts DiagnosticPod MasterConfigCheck MasterNode NodeConfigCheck NodeDefinitions UnitStatus
+# Without things feeding into systemd, AnalyzeLogs and UnitStatus are irrelevant.
+# ConfigContexts runs into some invalid contexts created by other tests, so is omitted.
```

Comment:
Got any pointers for cleaning? AIUI the kubeconfig is written at the beginning of all tests and there happens to be another test that pumps out a bunch of bogus projects or something like that (can't remember exactly).  Unless there is another test that needs to "reset" the kubeconfig I expect it goes against the test architecture a little bit.